crates/ra_mbe/src/lib.rs

```diff
@@ -42,6 +42,8 @@ pub use crate::syntax_bridge::{
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct MacroRules {
     pub(crate) rules: Vec<Rule>,
+    /// Highest id of the token we have in TokenMap
+    pub(crate) shift: Option<u32>,
```

Comment:
Why is this an option? We can always safely store `0` here.

Reply:
Because I thought the TokenId is started from zero and if there is not shift, we don't need to shift it. (Note that if shift is zero, it can mean a zero token-id is used in macro-rules).

But you are right, if the shift is zero, which means it is `() => {}`, which would not accept any arguments :)

Reply:
To be more precise, I think it makes sense for `compute_max_id` function to return an `Option`, as there may be no tokens with ids. The `shift` however should zero if ther are no tokens, so we should compute it like `max_id.map_or(0, |it| it + 1)`

Reply:
Fixed.